### PR TITLE
DIABLO-620, 'blackouts' schema with model, controller, and tests

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/DIABLO-620
https://jira.ets.berkeley.edu/jira/browse/DIABLO-621

Like the NYC blackout of '77.

image

### CODE REVIEW

**Comment by johncrossman** [comment]
[Diablo v1.2 release instructions](https://docs.google.com/document/d/1JtOs9G6tyvufhchvXPG-uBMJWucb2RBPZUKLAx91eJo/edit?usp=sharing) updated.

File: diablo/api/blackout_controller.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            start_date=_local_blackout_date_to_utc(start_date),
+            end_date=_local_blackout_date_to_utc(end_date),
+        )
+        return tolerant_jsonify(blackout.to_api_json())
+    else:
+        raise ResourceNotFoundError('No such email template')
+
+
+def _local_blackout_date_to_utc(blackout_date):
+    return localized_timestamp_to_utc(blackout_date, date_format='%Y-%m-%d')
```
Comment (pauline2k) [inline diablo/api/blackout_controller.py:101]: @johncrossman Should we raise a BadRequestError here if the user-supplied date param doesn't parse correctly?

File: diablo/api/blackout_controller.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    start_date = params.get('startDate')
+    end_date = params.get('endDate')
+
+    if None in [name, start_date, end_date]:
+        raise BadRequestError('Required parameters are missing.')
+
+    blackout = Blackout.create(
+        name=name,
+        start_date=_local_blackout_date_to_utc(start_date),
+        end_date=_local_blackout_date_to_utc(end_date),
```
Comment (pauline2k) [inline diablo/api/blackout_controller.py:70]: @johncrossman Also maybe a sanity check to ensure that end_date is equal to or later than start_date?

**Comment by pauline2k** [comment]
Looks good to me, @johncrossman. A couple questions about validating date params on the way in.

File: diablo/api/blackout_controller.py (hunk 1 above)
Comment (johncrossman) [reply diablo/api/blackout_controller.py:101]: @pauline2k - Yes, I agree.

**Comment by johncrossman** [comment]
Thanks @pauline2k  Amendments coming soon.

**Comment by johncrossman** [comment]
@pauline2k - Travis green after amendments.

**Pull request merged**